File: 2labz/2labz3/3.swift
import Foundation

print("Введите количество чисел: ", terminator: "")
guard let nStr = readLine(), let n = Int(nStr) else {
    fatalError("Ошибка ввода")
}

var result: [String] = []
for i in 1...n {
    print("Введите число \(i): ", terminator: "")
    guard let numStr = readLine(), let num = Int(numStr) else {
        fatalError("Ошибка ввода")
    }
    
    // Пропускаем числа от 1 до 9
    if (1...9).contains(num) {
        continue
    }
    
    // Переворачиваем число
    let reversed = Int(String(String(num).reversed()))!
    result.append(String(reversed))
}

print("Перевернутые числа без ведущих нулей: \(result.joined(separator: " "))")

File: 2labz/2labz3/3.js
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

readline.question('Введите количество чисел: ', (nStr) => {
    const n = parseInt(nStr);
    const result = [];
    let i = 0;
    
    const askNumber = () => {
        if (i >= n) {
            console.log(`Перевернутые числа без ведущих нулей: ${result.join(' ')}`);
            readline.close();
            return;
        }
        
        readline.question(`Введите число ${i+1}: `, (numStr) => {
            const num = parseInt(numStr);
            
            if (num >= 1 && num <= 9) {
                i++;
                askNumber();
                return;
            }
            
            const reversed = parseInt(numStr.split('').reverse().join(''));
            result.push(reversed.toString());
            i++;
            askNumber();
        });
    };
    
    askNumber();
});